pyomo/contrib/sensitivity_toolbox/sens.py

```diff
@@ -7,216 +7,684 @@
+              c2: x2 + x3 = p2
+              0 <= x1, x2, x3 <= 10
+              p1 = 10
+              p2 = 5
+    the function retuns dx/dp and dp/dp, and colum orders.
     Parameters
     ----------
-    instance: ConcreteModel
-        pyomo model object
+    model: Pyomo ConcreteModel
+        model should includes an objective function
+    theta_names: list of strings
+        List of Var names
+    theta: dict
+        Estimated parameters e.g) from parmest
+    tee: bool, optional
+        Indicates that ef solver output should be teed
+    solver_options: dict, optional
+        Provides options to the solver (also the name of an attribute)
+    var_dic: dictionary
+        If any original variable contains "'", need an auxiliary dictionary 
+        with keys theta_namess without "'", values with "'".
+        e.g) var_dic: {'fs.properties.tau[benzene,toluene]': "fs.properties.tau['benzene','toluene']",
+                       'fs.properties.tau[toluene,benzene]': "fs.properties.tau['toluene','benzene']"}
 
-    paramSubList: list
-        list of mutable parameters
+    Returns
+    -------
+    dsdp_dic: dict
+        gradient vector of the (decision variables, parameters) with respect to paramerters (=theta_name).
+        e.g) dict = {'d(x1)/d(p1)', 'd(x2)/d(p1)', 'd(p1)/d(p1)', 'd(p2)/d(p1)', 'd(x3)/d(p1)', 
+                     'd(x1)/d(p2)', 'd(x2)/d(p2)', 'd(p1)/d(p2)', 'd(p2)/d(p2)', 'd(x3)/d(p2)'},
+    col: list
+        list of variable names
+        e.g) col = ['x1', 'x2', 'p1', 'p2', 'x3'].
+    """
+    m = model.clone()
+    original_Param = []
+    perturbed_Param = []
+    m.extra = ConstraintList()
+    kk = 0
+    if var_dic == {}:
+        for i in theta_names:
+            var_dic[i] = i
+    for v in theta_names:
+        v_tmp = str(kk)
+        setattr(m, str('original_')+v_tmp ,Param(initialize=theta[v], mutable=True))
+        setattr(m, str('perturbed_')+v_tmp ,Param(initialize=theta[v]))
+        m.extra.add(eval('m.'+var_dic[v]) - eval('m.original_'+v_tmp) == 0 )
+        original_Param.append(eval('m.original_'+v_tmp))
+        perturbed_Param.append(eval('m.perturbed_'+v_tmp))
+        kk = kk + 1
+    m_kaug_dsdp = sensitivity_calculation('kaug',m,original_Param,perturbed_Param, tee)
 
-    perturbList: list
-        list of perturbed parameter values
+    try:
+        with open ("./dsdp/col_row.col", "r") as myfile:
+            col = myfile.read().splitlines()
+        dsdp = np.loadtxt("./dsdp/dsdp_in_.in")
+    except Exception as e:
+        print('File not found.')
 
-    cloneModel: bool, optional
-        indicator to clone the model. If set to False, the original
-        model will be altered
+    dsdp = dsdp.reshape((len(theta_names), int(len(dsdp)/len(theta_names))))
+    dsdp = dsdp[:len(theta_names), :len(col)]
+    dsdp_dic = {}
+    for i in range(len(theta_names)):
+        for j in range(len(col)):
+            if "_SENSITIVITY_TOOLBOX_DATA" not in col[j]:
```

Comment:
Should probably use `SensitivityInterface.get_default_block_name()` here instead of hard-coding the string, in case we change it later.

Reply:
Thanks for the suggestion, I updated it.